tests/unit/mathematics/SGLinalg_unittest.cc

```diff
@@ -60,6 +60,21 @@ TEST(SGLinalg, CPUBackend_dot)
 	EXPECT_NEAR(result, 20.0, 1E-15);
 }
 
+TEST(SGLinalg, SGVector_CPUBackend_dot)
+{
+	const index_t size=10;
+	SGVector<int32_t> a(size), b(size);
+	a.set_const(1);
```

Comment:
use range_fill please, not constant. Covers many more cases where things can go wrong